src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -113,84 +114,82 @@ public void Start<TContext>(IHttpApplication<TContext> application)
                 engine.Start(threadCount);
                 var atLeastOneListener = false;
 
-                foreach (var address in _serverAddresses.Addresses.ToArray())
-                {
-                    var parsedAddress = ServerAddress.FromUrl(address);
-                    atLeastOneListener = true;
+                var listenOptions = Options.ListenOptions;
 
-                    if (!parsedAddress.Host.Equals("localhost", StringComparison.OrdinalIgnoreCase))
+                if (listenOptions.Any())
+                {
+                    var addresses = _serverAddresses.Addresses;
+                    if (addresses.SingleOrDefault() != "http://localhost:5000")
                     {
-                        try
-                        {
-                            _disposables.Push(engine.CreateServer(parsedAddress));
-                        }
-                        catch (AggregateException ex)
-                        {
-                            if ((ex.InnerException as UvException)?.StatusCode == Constants.EADDRINUSE)
-                            {
-                                throw new IOException($"Failed to bind to address {parsedAddress}: address already in use.", ex);
-                            }
-
-                            throw;
-                        }
+                        var joined = string.Join(", ", addresses);
+                        _logger.LogWarning($"Address(es) '{joined}' were overridden. Binding to endpoints defined in UseKestrel instead.");
```

Comment:
Why would we do this instead of throwing? The only reason I can think of is if some component like IISIntegration set it up and then the user tried to override it by specifying the bindings directly?